lib/iis_shc_issuer/qbp_client.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+require 'savon'
+require 'ruby-hl7'
+require 'faraday'
+require_relative 'qpd'
+
+module IISSHCIssuer
+  # Send, receive, and translate HL7 V2 messages from the QBP client of the IIS sandbox
+  module QBPClient
+    extend self
+    # Query a patient's Immunization history from the IIS Sandbox
+    # @param patient_info [Hash] Patient demographic info sent from the IIS Consumer Portal
+    # @param credentials [Hash] (optional) User Id, Password, and Facility Id for the IIS Sandbox login
+    # @return [HL7::Message] The HL7 V2 Response message from the IIS-Sandbox
+    def query(patient_info,
+              sandbox_credentials = { username: Rails.application.config.username,
+                                      password: Rails.application.config.password,
+                                      facilityID: Rails.application.config.facilityID })
```

Comment:
So the key here can't change because the soap client actually looks for these specific elements when you make a request (see[ line 19 of the WSDL here](https://github.com/dvci/iis-shc-issuer/blob/qbp-client/lib/assets/service.wsdl#L19) for where that is defined). You can see on line 33 of this file that this hash is just spread into the `:submit_single_message` call, along with the actual HL7 message.

One option could be do take in a "`facility_id`" variable, and then reassign it to the `:facilityID` element within the actual SOAP message, but I don't personally see any benefit to doing that extra step.